test/dlc_tests/conftest.py

```diff
@@ -222,10 +235,112 @@ def pytest_runtest_setup(item):
+        test_coverage_file = test_utils.TEST_COVERAGE_FILE
+        test_cov = {}
+        for item in items:
+            # Define additional csv options
+            function_name = item.name.split("[")[0]
+            function_key = f"{item.fspath}::{function_name}"
+            str_fspath = str(item.fspath)
+            str_keywords = str(item.keywords)
+
+            # Based on keywords and filepaths, assign values
+            scope = infer_field_value("all", ("mxnet", "tensorflow", "pytorch"), str_fspath)
+            train_inf = infer_field_value("both", ("training", "inference"), str_fspath, str_keywords)
+            integration = infer_field_value("general integration",
+                                            ("_dgl_", "smdebug", "gluonnlp", "smexperiments", "_mme_", "pipemode",
+                                             "tensorboard", "_s3_"),
+                                            str_keywords)
+            model = infer_field_value("N/A",
+                                      ("mnist", "densenet", "squeezenet", "half_plus_two", "half_plus_three"),
+                                      str_keywords)
+            num_instances = infer_field_value(1, ("_multinode_", "_multi-node_"), str_fspath, str_keywords)
+            cpu_gpu = infer_field_value("all", ("cpu", "gpu", "eia"), str_keywords)
+            if cpu_gpu == "gpu":
+                if "p2.xlarge" in str_keywords:
+                    cpu_gpu = "single_gpu"
+
+            # Construct Category and Github_Link fields based on the filepath
+            category = str_fspath.split('/dlc_tests/')[-1].split('/')[0]
+            github_link = f"https://github.com/aws/deep-learning-containers/blob/master/" \
+                          f"{str_fspath.split('/deep-learning-containers/')[-1]}"
+
+            # Create a new test coverage item if we have not seen the function before. This is a necessary step,
+            # as parametrization can make it appear as if the same test function is a unique test function
+            if not test_cov.get(function_key):
+                test_cov[function_key] = {
+                                            "Category": category,
+                                            "Name": function_name,
+                                            "Scope": scope,
+                                            "Job_Type": train_inf,
+                                            "Num_Instances": get_marker_arg_value(item, "multinode", num_instances),
+                                            "Processor": cpu_gpu,
+                                            "Integration": get_marker_arg_value(item, "integration", integration),
+                                            "Model": get_marker_arg_value(item, "model", model),
+                                            "Github_Link": github_link,
+                                           }
+        with open(test_coverage_file, "w+") as tc_file:
```

Comment:
Let's separate this out into it's own function.

Reply:
sure, good call